### PR TITLE
chore: add important special block style

### PR DESCRIPTION
**Pull request by unknwon**
###### Important

Before:
CleanShot 2023-09-04 at 17 45 05@2x

After:

CleanShot 2023-09-04 at 19 17 10@2x

###### Warning

Before:
CleanShot 2023-09-04 at 19 17 49@2x

After:
CleanShot 2023-09-04 at 19 17 59@2x

###### Note

Before:
CleanShot 2023-09-04 at 19 18 37@2x

After:
CleanShot 2023-09-04 at 19 18 48@2x

### CODE REVIEW

**Review by michaellzc — COMMENTED** [review]
can we reuse github flavored markdown? https://github.com/orgs/community/discussions/16925

**Comment by unknwon** [comment]
> can we reuse github flavored markdown? https://github.com/orgs/community/discussions/16925

@michaellzc  You mean the markdown syntax part (`> [!IMPORTANT]`) and/or rendered style? (The former should be straightforward and only need global find-and-replace. Not sure how much work to make the latter work 😂

**Comment by michaellzc** [comment]
> > can we reuse github flavored markdown? https://github.com/orgs/community/discussions/16925
> 
> @michaellzc You mean the markdown syntax part (`> [!IMPORTANT]`) and/or rendered style? (The former should be straightforward and only need global find-and-replace. Not sure how much work to make the latter work 😂

Ideally, we can support the `> [!IMPORTANT]` syntax.

🤔 it shouldn't be too hard, I think? When traversing the tree, whenever we see `[!IMPORTANT]` at the beginning of the codeblock, just remove the first node `> [!IMPORTANT]` and return the rest of stuff

https://unifiedjs.com/learn/recipe/remove-node/

styling doesn't really matter to me, as long as we can tell the difference between NOTE/IMPORTANT/WARNING

**Comment by unknwon** [comment]
> Ideally, we can support the `> [!IMPORTANT]` syntax.

This part is actually easy IMO (supporting the syntax). Rendering exactly like the GitHub ones is unknown to me (CSS magic 😂 )

**Comment by unknwon** [comment]
@michaellzc - alright, this PR has from small QoL improvement to a controversial one, see https://github.com/sourcegraph/handbook/pull/7738/commits/26417559f3fcebda872c4639615fb0b874e13a94 and updated screenshots.

File: src/lib/markdownToHtml.ts
Comment (unknwon) [inline src/lib/markdownToHtml.ts:1]: This is the file that really matters

**Review by michaellzc — APPROVED** [review]
nice!

**Comment by unknwon** [comment]
OK, I hope people don't get mad at me 😛

**Comment by michaellzc** [comment]
> OK, I hope people don't get mad at me 😛

lol, it's much easier to follow a more well-adopted standard instead of inventing our own. 

https://github.com/remarkjs/remark-gfm will catch up soon

**Review by filiphaftek — COMMENTED** [review]
Awesome!

**Pull request merged**